sklearn/utils/_test_common/instance_generator.py

```diff
@@ -711,6 +716,38 @@
         ],
     },
     SkewedChi2Sampler: {"check_dict_unchanged": dict(n_components=1)},
+    SparseCoder: {
+        "check_estimators_dtypes": dict(dictionary=rng.normal(size=(5, 5))),
```

Comment:
is there a single case where we can set to have it pass all the tests?

Reply:
`dictionary` is not a very friendly parameter because it needs to have a shape compatible with `X`, but all the checks have different `X`s